service/s3/api_client.go

```diff
@@ -452,3 +452,125 @@ func addRequestResponseLogging(stack *middleware.Stack, o Options) error {
 		LogResponseWithBody: o.ClientLogMode.IsResponseWithBody(),
 	}, middleware.After)
 }
+
+// PresignOptions represents the presign client options
+type PresignOptions struct {
+	// ClientOptions are list of functional options to mutate client options used by
+	// presign client
+	ClientOptions []func(*Options)
+	// Presigner is the presigner used by the presign url client
+	Presigner v4.HTTPPresigner
+	// Expires sets the expiration duration for the generated presign url
+	Expires time.Duration
```

Comment:
This should call out that expires has a precision of seconds. Also if the field is not required, and the SDK shouldn't set a default, it should call out the default behavior if unset.